### PR TITLE
[Windows] Refactor WiX installers for Windows, Windows Engine and Developer to remove versioned paths and registry for simplified future upgrades

### PR DESCRIPTION
**Pull request by mcdurdin**
In this PR, I refactor a bunch of WiX code and try to make it easier to maintain as well as removing a lot of version-specific data which just makes it harder to do major upgrades. In the future, Keyman products on Windows will install into an unversioned path and use unversioned registry settings. Installed version may be detected by reading a registry value from `HKLM\Software\[Wow6432Node\]Keyman\..`.

### CODE REVIEW

**Comment by mcdurdin** [comment]
# Test Protocols

1. Clean install of 11.0.1500.0
2. Clean install of 22.0.900.0
3. Upgrade from 11.0.1500.0 to 11.0.1501.0
4. Upgrade from 11.0.1501.0 to 11.0.1503.0
5. Upgrade from 11.0.1503.0 to 22.0.900.0
6. Upgrade from 10.0 to 11.0.1500.0 to 11.0.1501.0
7. Upgrade from 9.0 to 10.0 to 11.0.1500.0 to 11.0.1501.0 to 22.0.900.0
8. Upgrade from 8.0 to 9.0 to 10.0 to 11.0.1500.0 to 11.0.1501.0 to 22.0.900.0
9. Upgrade from 8.0 to 11.0.1500.0 to 11.0.1501.0 to 22.0.900.0
9. Upgrade from 9.0 to 11.0.1500.0 to 11.0.1501.0 to 22.0.900.0

# Initial manual checks

1. Does Keyman run?
2. Any incorrect files / registry / version?
3. Are version numbers displayed correct?

# Checks at each new version

1. Are file versions correct in folders?
     C:\Program Files (x86)\Common Files\Keyman
     C:\Program Files (x86)\Keyman
     C:\ProgramData\Keyman
     %AppData%\Keyman
     C:\Program Files (x86)\Common Files\Tavultesoft
     C:\Program Files (x86)\Tavultesoft
     C:\ProgramData\Tavultesoft
     %AppData%\Tavultesoft
     
2. Are registry settings correct (cleaned older versions, new version keys correct)
     HKCU\Software\Keyman
     HKCU\Software\Tavultesoft
     HKLM\Software\Wow6432Node\Keyman
     HKLM\Software\Wow6432Node\Tavultesoft
     TSF registry keys?
     Others? (e.g. windows keyboards)
     
3. Are keyboard layouts in correct place and correctly registered?

4. Can we automate this -- for 11.0 onward installs?

# First run test notes

1. Install options shows reference to Keyman Desktop 10 (Upgrade Keyman Desktop 6, 7, 8 or 9 keyboards to Keyman Desktop 10)
2. Dialog during SETUP: Error checking for updates: Error 404
3. Dialog during SETUP: Keyman Desktop 11.0 Setup: Unable to find the Keyman Desktop directory. You should reinstall the product.
4. Keyman does install but due to above error, setup bootstrap window does not disappear
5. Keyman does not start:
```
[Window Title]
Error

[Content]
kmshell has had a fatal error.  An additional error was encountered starting the exception manager (The system cannot find the file specified).  
Error log is stored in 

  C:\Users\mcdurdin\AppData\Roaming\Keyman\Diag\exception-7024-2018-09-13-21-09-34.log

Exception EKeymanPath in kmshell.exe: Unable to find the Keyman Desktop directory.  You should reinstall the product.

Please send this information to Keyman Support

[OK]
```

**Comment by mcdurdin** [comment]
Attached [results.zip](https://github.com/keymanapp/keyman/files/2387173/results.zip) is the full test suite results for Keyman upgrades from 9.0, 10.0 and 11.0 onwards. The only issue discovered was that a faked "version 22.0" shows `HKLM\Software\Wow6432Node\Keyman\Keyman Desktop`, `release = "11.0"` instead of `"22.0"`

Once that is resolved, this should be ready for review and merge.

**Comment by mcdurdin** [comment]
Last 3 commits resolve the issue with the incorrect release number in registry. This should be ready for review now.

File: windows/src/global/delphi/general/keymanversion.inc [hunk 1]
```diff
@@ -0,0 +1 @@
+  SKeymanVersion = '11.0';
```
Comment (darcywong00) [inline windows/src/global/delphi/general/keymanversion.inc]: Is it possible to combine this with `windows/src/version.txt`?
Or will this need to be updated every major release?

Comment (mcdurdin) [reply windows/src/global/delphi/general/keymanversion.inc]: Ah, that was not supposed to be added -- missed that, sorry. It's gone now.

**Pull request merged**